date_range/models/date_range_search_mixin.py

```diff
@@ -20,9 +25,35 @@ class DateRangeSearchMixin(models.AbstractModel):
         search="_search_date_range_search_id")
 
     def _compute_date_range_search_id(self):
-        """Assign a dummy value for this search field"""
+        """Assign a value for the computed field.
+
+        This can either a matching date range from the specified type,
+        if any, or False.
+        """
+        record2range = {}
+        if self._date_range_assign_type_code:
+            dr_type = self.env["date.range.type"].search(
+                [("code", "=", self._date_range_assign_type_code),
+                 ("company_id", "=", False)], limit=1)
+            if dr_type:
+                query = sql.SQL(
+                    """
+                    SELECT tbl.id, dr.id
+                    FROM {} AS tbl
+                    JOIN date_range AS dr
+                        ON {} BETWEEN dr.date_start AND dr.date_end
+                    WHERE dr.type_id = %s AND tbl.id IN %s
+                    """).format(
+                        sql.Identifier(self._table),
+                        sql.Identifier(self._date_range_search_field),
+                    )
+                self.env.cr.execute(
+                    query,
+                    (dr_type.id, tuple(self.ids or [0])),
```

Comment:
this can be more intelligible with [AsIs](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.AsIs)

Reply:
What do you mean exactly?

Reply:
```python
query,
(dr_type.id, tuple(self.ids or [0]), AsIs(table), etc),
```

Reply:
If you mean, use `AsIs` instead of `sql.SQL`, I believe `AsIs` does not really help against SQL injection so I think `sql.SQL` should be used as much as possible for cases like these.

Reply:
you don't need to protect against sql injection from developers using your code

Reply:
Like I said, it's good practice.